src/main/java/core/basesyntax/model/Operation.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.model;
+
+import core.basesyntax.exception.NoSuchOperationException;
+
+public enum Operation {
+    BALANCE("b"),
+    SUPPLY("s"),
+    PURCHASE("p"),
+    RETURN("r");
+
+    private static final String NO_SUCH_OPERATION_MESSAGE =
+            "This operation is not in the supported list: ";
+    private final String code;
+
+    Operation(String code) {
+        this.code = code;
+    }
+
+    public String getCode() {
+        return code;
+    }
+
+    public static Operation fromString(String operationInString) {
+        for (Operation operation : Operation.values()) {
+            if (operation.code.equalsIgnoreCase(operationInString)) {
+                return operation;
+            }
+        }
```

Comment:
I think it's better to use "switch-case" here

Reply:
Maybe the switch case will be faster, but I think for each looks better and can potentially save us from inserting new "case" blocks into the switch construction.

Reply:
*need mentor help)

Reply:
There are 2 good ways :)